src/include/cpu/rpp_cpu_simd.hpp

```diff
@@ -885,4 +894,477 @@ static inline void fast_matmul4x4_sse(float *A, float *B, float *C)
+        *(T1[2] + cnt) = (Rpp32f) *(srcPtrTopRow + loc[2] + cnt);
+        *(T1[3] + cnt) = (Rpp32f) *(srcPtrTopRow + loc[3] + cnt);
+        *(B1[0] + cnt) = (Rpp32f) *(srcPtrBottomRow + loc[0] + cnt);
+        *(B1[1] + cnt) = (Rpp32f) *(srcPtrBottomRow + loc[1] + cnt);
+        *(B1[2] + cnt) = (Rpp32f) *(srcPtrBottomRow + loc[2] + cnt);
+        *(B1[3] + cnt) = (Rpp32f) *(srcPtrBottomRow + loc[3] + cnt);
+    }
+    __m128 pTemp[8];
+    pTemp[0] = _mm_loadu_ps(T1[0]);
+    pTemp[1] = _mm_loadu_ps(T1[1]);
+    pTemp[2] = _mm_loadu_ps(T1[2]);
+    pTemp[3] = _mm_loadu_ps(T1[3]);
+    pTemp[0] = _mm_unpacklo_ps(pTemp[0], pTemp[2]);
+    pTemp[1] = _mm_unpacklo_ps(pTemp[1], pTemp[3]);
+    p[0] = _mm_unpacklo_ps(pTemp[0], pTemp[1]);
+    p[1] = _mm_unpackhi_ps(pTemp[0], pTemp[1]);
+    pTemp[0] = _mm_loadu_ps(B1[0]);
+    pTemp[1] = _mm_loadu_ps(B1[1]);
+    pTemp[2] = _mm_loadu_ps(B1[2]);
+    pTemp[3] = _mm_loadu_ps(B1[3]);
+    pTemp[0] = _mm_unpacklo_ps(pTemp[0], pTemp[2]);
+    pTemp[1] = _mm_unpacklo_ps(pTemp[1], pTemp[3]);
+    p[2] = _mm_unpacklo_ps(pTemp[0], pTemp[1]);
+    p[3] = _mm_unpackhi_ps(pTemp[0], pTemp[1]);
+
+    return RPP_SUCCESS;
+}
+
+inline RppStatus rpp_store4_f32pln3_to_f16pln3(Rpp16f* dstRPtr, Rpp16f* dstGPtr, Rpp16f* dstBPtr, __m128* p)
+{
+    float temp[3][4];
+    _mm_storeu_ps((float *)temp[0], p[0]);
+    _mm_storeu_ps((float *)temp[1], p[1]);
+    _mm_storeu_ps((float *)temp[2], p[2]);
+    /* Convert float pixels to float16 type and store in destination */
+    for(int i = 0; i < 4; i++)
+    {
+        *(dstRPtr + i) = (Rpp16f)(temp[0][i]);
+        *(dstGPtr + i) = (Rpp16f)(temp[1][i]);
+        *(dstBPtr + i) = (Rpp16f)(temp[2][i]);
+    }
+
+    return RPP_SUCCESS;
+}
+
+inline RppStatus rpp_store4_f32pln3_to_f16pkd3(Rpp16f* dstPtr, __m128* p)
+{
+    __m128 pTemp[7];
+    float temp[4][4];
+    pTemp[0] = _mm_unpacklo_ps(p[0], p[1]);
+    pTemp[1] = _mm_unpackhi_ps(p[0], p[1]);
+    pTemp[2] = _mm_shuffle_ps(pTemp[0], p[2], _MM_SHUFFLE(0, 0, 1, 0));
+    pTemp[3] = _mm_shuffle_ps(pTemp[0], p[2], _MM_SHUFFLE(0, 1, 3, 2));
+    pTemp[4] = _mm_shuffle_ps(pTemp[1], p[2], _MM_SHUFFLE(0, 2, 1, 0));
+    pTemp[5] = _mm_shuffle_ps(pTemp[1], p[2], _MM_SHUFFLE(0, 3, 3, 2));
+    _mm_storeu_ps((float *)temp[0], pTemp[2]);
+    _mm_storeu_ps((float *)temp[1], pTemp[3]);
+    _mm_storeu_ps((float *)temp[2], pTemp[4]);
+    _mm_storeu_ps((float *)temp[3], pTemp[5]);
+    /* Convert float pixels to float16 type and store in destination */
+    for(int i = 0; i < 3; i++)
+    {
+        *(dstPtr + i) = (Rpp16f)(temp[0][i]);
+        *(dstPtr + 3 + i) = (Rpp16f)(temp[1][i]);
+        *(dstPtr + 6 + i) = (Rpp16f)(temp[2][i]);
+        *(dstPtr + 9 + i) = (Rpp16f)(temp[3][i]);
+    }
+
+    return RPP_SUCCESS;
+}
+
+inline void compute_resize_src_loc_sse(__m128 &dstLoc, __m128 &scale, __m128 &limit, Rpp32u *srcLoc, __m128 &weight, bool hasRGBChannels = false)
```

Comment:
Lets move all these compute functions to rpp_cpu_common under the compute section. Use the rpp_cpu_simd only for generalized load/store routine pairs.

Reply:
Done